store/index.js

```diff
@@ -172,14 +172,33 @@ export const getters = {
         return mappedColumns;
     },
 
-    getMissingValues: (p_state) => (p_column) => {
+    getMissingValues: (p_state) => (p_category) => {
 
-        let missingValues = [];
+        // 0. Retrieve all columns linked with the given category
+        const mappedColumns = [];
+        for ( const column in p_state.columnToCategoryMapping ) {
+
+            if ( p_category === p_state.columnToCategoryMapping[column] ) {
+
+                mappedColumns.push(column);
+            }
+        }
+
+        // 1. Build a map of missing values by column
+        let missingValues = {};
+        for ( const column of mappedColumns ) {
+
+            // A. Starts out as a blank list
+            missingValues[column] = [];
 
-        if ( p_column in p_state.dataDictionary.annotated &&
-             "missingValues" in p_state.dataDictionary.annotated[p_column] ) {
+            // B. Save a list of missing values for this column if,
+            // 1) the column has an entry in the data dictionary and,
+            // 2) if a missing values list for the column has already been made
+            if ( column in p_state.dataDictionary.annotated &&
+                "missingValues" in p_state.dataDictionary.annotated[column] ) {
 
-            missingValues = p_state.dataDictionary.annotated[p_column].missingValues;
+               missingValues[column] = p_state.dataDictionary.annotated[column].missingValues;
```

Comment:
The `annot-missing-values` component expects a different format atm, see: https://github.com/neurobagel/annotation_tool/blob/dev_components_talk_to_store_directly/components/annot-missing-values.vue#L80 (i.e. array of objects for each value, object has column name and value repeated).

Whatever format you pick, make sure that the format the getter produces matches what the component expects.

Reply:
@surchs After looking the code over, I believe this implementation is consistent with what the `annot-missing-values` component expects in its `tableItems` function. The only change that needs to occur for `annot-missing-values` is the name change for the getter.

That said, in looking over the `tableItems` function there I'd like to refactor it. The code inside is quite confusing for me to read. I just did so in order to better understand what it's attempting to do - comment description aside. I can add that refactor in a separate PR or here since an adjustment needs to be made to `annot-missing-values` anyway.

Reply:
I'm just going to make the name change for the getting in `annot-missing-values` and will include a refactor for `tableItems` in a separate PR.